homeassistant/components/minecraft_server/__init__.py

```diff
@@ -0,0 +1,428 @@
+
+            self._version = status_response.version.name
+            self._protocol_version = status_response.version.protocol
+            self._players_online = status_response.players.online
+            self._players_max = status_response.players.max
+            self._latency = status_response.latency
+
+            if status_response.players.sample is None:
+                self._players_list = "[]"
+            else:
+                self._players_list = "["
+
+                for player in status_response.players.sample:
+                    self._players_list += player.name + ", "
+
+                # Remove last seperator ", " and add end bracket.
+                self._players_list = self._players_list[:-2] + "]"
+
+                # Limit players list length to 255.
+                if len(self._players_list) > 255:
+                    self._players_list = self._players_list[:-4] + "...]"
+                    _LOGGER.debug("Players list length > 255 (truncated).")
+
+    def _print_data(self):
+        """Print properties."""
+        _LOGGER.debug(
+            """Properties of '%s' (%s:%s):
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s""",
+            self._name,
+            self._host,
+            self._port,
+            NAME_DESCRIPTION,
+            self._description,
+            NAME_VERSION,
+            self._version,
+            NAME_PROTOCOL_VERSION,
+            self._protocol_version,
+            NAME_LATENCY_TIME,
+            self._latency,
+            NAME_PLAYERS_ONLINE,
+            self._players_online,
+            NAME_PLAYERS_MAX,
+            self._players_max,
+            NAME_PLAYERS_LIST,
+            self._players_list,
+        )
+
+
+class MinecraftServerEntity(Entity):
+    """Representation of a Minecraft Server base entity."""
+
+    def __init__(self, hass, server, name, unit, icon):
+        """Initialize base entity."""
+        self._server = server
+        self._hass = hass
+        self._state = None
+        self._name = server.name() + " " + name
+        self._sensor_name = name
+        self._unit = unit
+        self._icon = icon
+
+        # Subscribe to signal from server instance.
+        signal_name_suffix = self._server.name()
+        signal_name_suffix = signal_name_suffix.replace(" ", "_")
+        signal_name_suffix = signal_name_suffix.lower()
+        self._signal_name = SIGNAL_NAME_PREFIX + signal_name_suffix
+        async_dispatcher_connect(
+            self._hass, self._signal_name, self._async_trigger_update
+        )
+
+    async def _async_trigger_update(self):
+        """Triggers update of properties after receiving signal from server instance."""
+        _LOGGER.debug("%s: Received signal '%s'.", self._sensor_name, self._signal_name)
+        self.async_schedule_update_ha_state(force_refresh=True)
+
```

Comment:
Add a `unique_id` so that your entries appreas in the UI _integrations -> Minecraft Server_

Depending on host so we can change the name in the future.

```python
    @property
    def unique_id(self) -> str:
        """Return a unique ID."""
        return f"{self._server.host}_{self._sensor_name}"
```

Reply:
I'll have a look at it and come back to you.

Reply:
Done.